src/System.Windows.Forms/src/System/Windows/Forms/ListView.ListViewAccessibleObject.cs

```diff
@@ -0,0 +1,240 @@
+                    UiaCore.NavigateDirection.FirstChild => GetChild(0),
+                    UiaCore.NavigateDirection.LastChild => GetChild(childCount - 1),
+                    _ => base.FragmentNavigate(direction)
+                };
+            }
+
+            internal override bool CanSelectMultiple => true;
+
+            internal override object GetPropertyValue(UiaCore.UIA propertyID)
+                => propertyID switch
+                {
+                    UiaCore.UIA.NamePropertyId
+                        => Name,
+                    UiaCore.UIA.AutomationIdPropertyId
+                        => OwningListView.IsHandleCreated ? OwningListView.Name : string.Empty,
+                    UiaCore.UIA.RuntimeIdPropertyId
+                        => RuntimeId,
+                    UiaCore.UIA.ControlTypePropertyId
+                        => UiaCore.UIA.ListControlTypeId,
+                    UiaCore.UIA.IsMultipleViewPatternAvailablePropertyId
+                        => IsPatternSupported(UiaCore.UIA.MultipleViewPatternId),
+                    UiaCore.UIA.ItemStatusPropertyId
+                        => GetItemStatus(),
+                    _ => base.GetPropertyValue(propertyID)
+                };
+
+            internal override int[] RuntimeId
+            {
+                get
+                {
+                    if (!OwningListView.IsHandleCreated)
+                    {
+                        return base.RuntimeId;
+                    }
+
+                    var runtimeId = new int[2];
+                    runtimeId[0] = 0x2a;
+                    runtimeId[1] = (int)(long)OwningListView.Handle;
+                    return runtimeId;
+                }
+            }
+
+            private string GetItemStatus()
+                => OwningListView.Sorting switch
+                {
+                    SortOrder.None => SR.NotSortedAccessibleStatus,
+                    SortOrder.Ascending => SR.SortedAscendingAccessibleStatus,
+                    SortOrder.Descending => SR.SortedDescendingAccessibleStatus,
+                    _ => null
+                };
+
+            public override AccessibleObject GetChild(int index)
+            {
+                if (index < 0 || index >= OwningListView.Items.Count)
+                {
+                    return null;
+                }
+
+                return new ListViewItemAccessibleObject(OwningListView, OwningListView.Items[index]);
+            }
+
+            public override int GetChildCount()
+            {
+                return OwningListView.IsHandleCreated ? OwningListView.Items.Count : 0;
+            }
+
+            internal override UiaCore.IRawElementProviderSimple[] GetSelection()
+            {
+                var selectedItemProviders = new List<UiaCore.IRawElementProviderSimple>();
+
+                if (!OwningListView.IsHandleCreated)
+                {
+                    return selectedItemProviders.ToArray();
+                }
```

Comment:
Unnecessary allocations
```suggestion
                if (!OwningListView.IsHandleCreated)
                {
                    return Array.Empty<UiaCore.IRawElementProviderSimple>();
                }

                var selectedItemProviders = new List<UiaCore.IRawElementProviderSimple>();
```

Reply:
Why can allocate `UiaCore.IRawElementProviderSimple[]` instead of a `List<UiaCore.IRawElementProviderSimple>()` and reduce on allocations.

Applies universally.

Reply:
Yes, agree, thanks, we can also use Linq `Select` in some places but need to include reference to Linq.